### PR TITLE
feat: vite 3

### PR DESCRIPTION
**Pull request by high1**
- updated to the latest vite version
- removed terser option and cleaned up vite config
- updated vite-plugin-solid and vite-plugin-pwa version
- updated pnpm-lock.yaml and yarn.lock files
- added a break between videos in the newest july 2022 blog article
- make routeReadyState more dry
- cleaned up editorconfig - added .mdx to extensions, removed redundant
options already declared globally
- prettier updated Resources formatting during the build

### CODE REVIEW

**Comment by davedbase** [comment]
Woah, this is an awesome PR! I'm excited to get us going on V3. This looks stable and works well for me. @modderme123 and @aquaductape if you want to have a look I'm ready to merge this in when you are.

**Comment by modderme123** [comment]
I think we should drop yarn.lock, but this PR looks good

File: package.json [hunk 1]
```diff
@@ -43,7 +43,6 @@
     "solid-dismiss": "^1.2.1",
     "solid-heroicons": "^2.0.3",
     "solid-js": "1.4.7",
-    "solid-jsx": "^0.9.0",
```
Comment (modderme123) [inline package.json:46]: I thought this was used by the mdx compiler...

File: src/pages/Articles/state-of-solid-july-2022.mdx [hunk 2]
```diff
@@ -89,6 +89,7 @@ Also have a read through some of these articles:
 And some fantastic video clips we enjoyed:
 
 <YouTube youTubeId="A_dUsSzxwkI" />
+<br />
```
Comment (modderme123) [inline src/pages/Articles/state-of-solid-july-2022.mdx:92]: maybe it is better to just have an extra line here and let md insert the linebreak?

File: package.json (hunk 1 above)
Comment (high1) [reply package.json:46]: It's not - solid-mdx is. This is unneeded.

File: src/pages/Articles/state-of-solid-july-2022.mdx (hunk 2 above)
Comment (high1) [reply src/pages/Articles/state-of-solid-july-2022.mdx:92]: Hm, I'm not sure how this works in mdx, I don't think that the space or empty rows after a component will be turned to linebreaks, but let's see...

**Comment by high1** [comment]
> I think we should drop yarn.lock but we can do that after merging

I can do that, I would also suggest that we add https://nodejs.org/api/packages.html#packagemanager property to package.json and set it to pnpm@7.8.0. It will benefit https://nodejs.org/api/corepack.html users for now. The only bad thing here is that it's not possible to declare it like pnpm@7 - the full version must be added, and this will require frequent updating...

**Comment by davedbase** [comment]
Let me know when we're ready to merge. I tested locally and it seems stable :-)

**Comment by davedbase** [comment]
I'm just going to do it, this looks fine to me aside. Feel free to make adjustments on main directly for the items you mentioned)

**Pull request merged**